destination/destination.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright © 2022 Meroxa, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package destination
+
+import (
+	"context"
+	"sync"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/conduitio/conduit-connector-zendesk/zendesk"
+)
+
+type Writer interface {
+	Write(ctx context.Context, records []sdk.Record) error
+}
+
+type Destination struct {
+	sdk.UnimplementedDestination
+	cfg          Config        // destination specific config for zendesk
+	buffer       []sdk.Record  // buffer stores the list of zendesk ticket from conduit server
+	ackFuncCache []sdk.AckFunc // returns error to conduit if fails else return nil
+	err          error
```

Comment:
What's this error?

Reply:
handle errors for the destination method implementation. done

Reply:
We should have a comment explaining what that error is: is it any error, or just the last write error etc.